### PR TITLE
Add file and line arguments to encoder/decoder test callbacks for better debug info

### LINKED ISSUE #14331: Resolve or remove TODO 3.0 in test/endecode_test.c
This is an RFE for better diagnostic output.
```
/*
 * TODO(3.0) For better error output, changed the callbacks to take __FILE__
 * and __LINE__ as first two arguments, and have them use the lower case
 * functions, such as test_strn_eq(), rather than the uppercase macros
 * (TEST_strn2_eq(), for example).
 */

```

There is a typo in the comment `change(d)`.

### PR DESCRIPTION
**Pull request by jon-oracle**
Added some extra macros locally to make it easier to call the test_* macros which stringize some arguments.

Fixes #14331

- [x] tests are added or updated

### CODE REVIEW

File: test/endecode_test.c [hunk 1]
```diff
@@ -25,6 +25,18 @@
...
+#define TEST_FL_mem_eq(a, m, b, n)   test_mem_eq(file, line, #a, #b, a, m, b, n)
+#define TEST_FL_strn_eq(a, b, n)     test_strn_eq(file, line, #a, #b, a, n, b, n)
+#define TEST_FL_strn2_eq(a, m, b, n) test_strn_eq(file, line, #a, #b, a, m, b, n)
+#define TEST_FL_int_eq(a, b)         test_int_eq(file, line, #a, #b, a, b)
+#define TEST_FL_int_ge(a, b)         test_int_ge(file, line, #a, #b, a, b)
+#define TEST_FL_int_gt(a, b)         test_int_gt(file, line, #a, #b, a, b)
+#define TEST_FL_long_gt(a, b)        test_long_gt(file, line, #a, #b, a, b)
+#define TEST_FL_true(a)              test_true(file, line, #a, (a) != 0)
+
+
```
Comment (t8m) [inline test/endecode_test.c:39]: Nit: please drop this extra empty line

**Review by t8m — COMMENTED** [review]
Apart from one nit, this LGTM.

File: test/endecode_test.c (hunk 1 above)
Comment (jon-oracle) [reply test/endecode_test.c]: Done

**Review by t8m — APPROVED** [review]

**Review by slontis — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
I've slightly edited the commit message to split subject and body of it and merged it to master. Thank you for your contribution.

**Pull request closed without merging**